src/js/App/Sidenav/Navigation/DynamicNav.js

```diff
@@ -20,11 +20,25 @@ const DynamicNav = ({ dynamicNav }) => {
     if (navigation) {
       if (typeof navigation === 'function') {
         Promise.resolve(navigation({ schema, dynamicNav, currentNamespace })).then((data) => {
-          schema.navItems = schema?.navItems?.map((item, key) => ({
-            ...item,
-            ...(key === schema.navItems.findIndex((nav) => nav.dynamicNav === dynamicNav) && data),
-          }));
-          dispatch(loadLeftNavSegment(schema, currentNamespace, window.location.pathname));
+          const indexOfDynamicNav = schema.navItems.findIndex((nav) => nav.dynamicNav === dynamicNav);
+          delete schema.navItems[indexOfDynamicNav].dynamicNav;
```

Comment:
You do not guard against a negative index (if it's not found)... is that possible?

Reply:
Well the dynamic nav brought us here, but I like defensive programming, so let's add it here!